code/modules/vehicles/mecha/mecha_construction_paths.dm

```diff
@@ -252,7 +289,100 @@
 		)
 	)
 
+/// Steps must be checked for sequentially, in case we skip into another state we don't want
+/// However, if we're doing a deconstruction step (backwards) we need to check in a different order
+/datum/component/construction/mecha/proc/skip_extra_steps(diff, forward)
+	var/on_valid_step = FALSE
+
+	while (!on_valid_step)
+		var/next_step = forward ? message_step : (message_step - 1)
```

Comment:
why `(message_step - 1)` here exactly? I don't really understand.

Reply:
Construction code is funky when it comes to messages. I had to write it out a couple times before I got it.

Let's ignore the treads step for this example.

We start on step 1. We haven't changed any numbers yet, and we're moving forward in construction, so we tell the user that they've connected the hydraulics systems, and advance to step 2.

Now if we want to disconnect the hydraulics systems, we can't print and then decrement the number; that would tell the user that they've deactivated the hydraulics instead of disconnecting them. That's why we have to offset the backwards step by one, in order to condense the code displaying messages.

Otherwise you'd get something like the old code:

```
if(2)
	if(diff==FORWARD)
		user.visible_message(span_notice("[user] activates [parent] hydraulic systems."), span_notice("You activate [parent] hydraulic systems."))
	else
		user.visible_message(span_notice("[user] disconnects [parent] hydraulic systems."), span_notice("You disconnect [parent] hydraulic systems."))
```

Reply:
I'll make a better comment for it in code